cmd/repo-query/repo-query.go

```diff
@@ -0,0 +1,353 @@
+
+	return cmd
+}
+
+func getOwnerAndRepo(url string) (owner, repo string) {
+	// Remove the "https://github.com/" prefix from the URL
+	url = strings.TrimPrefix(url, "https://github.com/")
+
+	// Split the remaining URL path into segments
+	segments := strings.Split(url, "/")
+
+	// The first segment is the owner, and the second segment is the repository name
+	if len(segments) >= 2 {
+		owner = segments[0]
+		repo = segments[1]
+	} else {
+		// URL is not in the expected format
+		owner = ""
+		repo = ""
+	}
+
+	return owner, repo
+}
+
+func run(opts *Options) error {
+	// get repo name and owner name from URL
+	owner, repo := getOwnerAndRepo(opts.URL)
+
+	fmt.Printf("Checking if %s/%s is indexed by us...⏳\n", owner, repo)
+	resp, err := http.Get(fmt.Sprintf("%s/collection?owner=%s&name=%s&branch=%s", repoQueryURL, owner, repo, opts.branch))
+	if err != nil {
+		return err
+	}
+
+	// not found or ok or error
+	if resp.StatusCode == http.StatusNotFound {
+		fmt.Println("Repo not found❗")
+		fmt.Println("Indexing repo...⏳")
+		err := indexRepo(owner, repo, opts.branch)
+		if err != nil {
+			return err
+		}
+
+		for {
+			fmt.Printf("\nWant to ask a question about %s/%s?\n", owner, repo)
+			fmt.Printf("> ")
+			// read input
+			scanner := bufio.NewScanner(os.Stdin)
+			if scanner.Scan() {
+				input := scanner.Text()
+				err := askQuestion(input, owner, repo, opts.branch)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else if resp.StatusCode == http.StatusOK {
+		fmt.Println("Repo found ✅")
+
+		// this should be an infinite loop
+		for {
+			fmt.Printf("\nWant to ask a question about %s/%s?\n", owner, repo)
+			fmt.Printf("> ")
+			// read input
+			scanner := bufio.NewScanner(os.Stdin)
+			if scanner.Scan() {
+				input := scanner.Text()
+				err := askQuestion(input, owner, repo, opts.branch)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("An error occurred: %v\n", string(body))
+		return errors.New("error while checking if repo is indexed")
+	}
+}
+
+type indexPostRequest struct {
+	Owner  string `json:"owner"`
+	Name   string `json:"name"`
+	Branch string `json:"branch"`
+}
+
+func indexRepo(owner string, repo string, branch string) error {
+	indexPostReq := &indexPostRequest{
+		Owner:  owner,
+		Name:   repo,
+		Branch: branch,
+	}
+
+	indexPostJSON, err := json.Marshal(indexPostReq)
+	if err != nil {
+		return err
+	}
+
+	responseBody := bytes.NewBuffer(indexPostJSON)
+	resp, err := http.Post(fmt.Sprintf("%s/embed", repoQueryURL), "application/json", responseBody)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("An error occurred: %v\n", string(body))
+		return errors.New("error while indexing repository")
```

Comment:
The error handling throughout could be abit more structured. I'd suggest using `fmt.Errorf` and returning a detailed error:

```suggestion
		if err != nil {
			return fmt.Errorf("error reading response body: %s", err.Error())
		}

		return fmt.Errorf("error while indexing repository: %s", err.Error())
```